src/averageAges.js

```diff
@@ -37,7 +39,14 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => person.sex === 'f');
+  const womenChildren = withChildren
+    ? women.filter(woman => people.some(person => person.mother === woman.name))
+    : women;
+  const womenAges = womenChildren.map(person => person.died - person.born);
+  const sumAges = womenAges.reduce((sum, age) => sum + age, 0);
+
+  return sumAges / womenAges.length;
```

Comment:
create a separate helper function to calculate average ages
you do the same three times